cloud/openstack/machineactuator.go

```diff
@@ -0,0 +1,420 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package openstack
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"os"
+	"reflect"
+	"strings"
+
+	"github.com/golang/glog"
+	"gopkg.in/yaml.v2"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"sigs.k8s.io/cluster-api/cloud/openstack/clients"
+	"sigs.k8s.io/cluster-api/cloud/openstack/machinesetup"
+	openstackconfigv1 "sigs.k8s.io/cluster-api/cloud/openstack/openstackproviderconfig/v1alpha1"
+	apierrors "sigs.k8s.io/cluster-api/errors"
+	"sigs.k8s.io/cluster-api/kubeadm"
+	clusterv1 "sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
+	client "sigs.k8s.io/cluster-api/pkg/client/clientset_generated/clientset/typed/cluster/v1alpha1"
+	"sigs.k8s.io/cluster-api/util"
+	"time"
+)
+
+const (
+	MachineSetupConfigPath   = "/etc/machinesetup/machine_setup_configs.yaml"
+	SshPrivateKeyPath        = "/etc/sshkeys/private"
+	SshPublicKeyPath         = "/etc/sshkeys/public"
+	SshKeyUserPath           = "/etc/sshkeys/user"
+	CloudConfigPath          = "/etc/cloud/cloud_config.yaml"
```

Comment:
The secret is defined as
```
secretName: cloud-config
```

So the CloudConfigPath should be `/etc/cloud/cloud-config`

Reply:
opps, thanks.